log-collector/pkg/fluentd/fluentd.go

```diff
@@ -0,0 +1,168 @@
+	}
+	if err := createMasterDeploy(client, namespace); err != nil {
+		return fmt.Errorf("error creating fluentd asset %v", err)
+	}
+	if err := createMasterSvc(client, namespace); err != nil {
+		return fmt.Errorf("error creating fluentd asset %v", err)
+	}
+
+	if err := createWorkerCfg(client, namespace); err != nil {
+		return fmt.Errorf("error creating fluentd asset %v", err)
+	}
+	if err := createWorkerDs(client, namespace); err != nil {
+		return fmt.Errorf("error creating fluentd asset %v", err)
+	}
+
+	if err := utils.Retry(10, time.Second*10, waitPodReadyOnMaster(client, namespace)); err != nil {
+		return err
+	}
+	if err := utils.Retry(10, time.Second*10, waitPodReadyOnWorker(client, namespace)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteAssets deletes all the fluentd assets.
+// Also removes all node annotations
+func DeleteAssets(client kubernetes.Interface, namespace string) error {
+	if err := deleteMasterCfg(client, namespace); err != nil {
+		return fmt.Errorf("error deleting fluentd asset %v", err)
+	}
+	if err := deleteMasterDeploy(client, namespace); err != nil {
+		return fmt.Errorf("error deleting fluentd asset %v", err)
+	}
+	if err := deleteMasterSvc(client, namespace); err != nil {
+		return fmt.Errorf("error deleting fluentd asset %v", err)
+	}
+
+	if err := deleteWorkerCfg(client, namespace); err != nil {
+		return fmt.Errorf("error deleting fluentd asset %v", err)
+	}
+	if err := deleteWorkerDs(client, namespace); err != nil {
+		return fmt.Errorf("error deleting fluentd asset %v", err)
+	}
+
+	labelSelc := fmt.Sprintf("%s=", fluentMasterNodeAnnotation)
+	nl, err := client.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: labelSelc})
+	if err != nil || len(nl.Items) == 0 {
+		return fmt.Errorf("couldn't find master node running fluent master %v", err)
+	}
+	n := nl.Items[0]
+	delete(n.ObjectMeta.Labels, fluentMasterNodeAnnotation)
+	if _, err := client.CoreV1().Nodes().Update(&n); err != nil {
+		return fmt.Errorf("couldn't delete annotation from the master node running fluent master %v", err)
+	}
+
+	return nil
+}
+
+// GetNodeAddressWithMaster return PublicAddressableIP of the master node that is running fluent master
+func GetNodeAddressWithMaster(client kubernetes.Interface, namespace string) (string, error) {
+	pl, err := client.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: "k8s-app=fluentd-master,tier=control-plane"})
+	if err != nil || len(pl.Items) == 0 {
+		return "", fmt.Errorf("No pod from fluentd-master deployment found: %v", err)
+	}
+
+	p := &pl.Items[0]
+	nodeName := p.Spec.NodeName
+
+	n, err := client.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("No node found with name: %s", nodeName)
+	}
+
+	var host string
+	for _, addr := range n.Status.Addresses {
+		if addr.Type == v1.NodeExternalIP {
+			host = addr.Address
+		}
+
+	}
+	if host == "" {
+		// try finding "LegacyHostIP"
```

Comment:
bootkube vagrant setups have LegacyHostIP as public addressable address in Node object